File: ios/Pets/Models/Sources/Models/Pet.swift
//
//  File.swift
//  
//
//  Created by Dylan Velez on 11/08/21.
//

import Foundation

public typealias Pets = [Pet]
public struct Pet: Codable {
    public let id: Int
    public let name: String
    public let breed: String
    public let sex: String
    public let age: Int
    public let color: String
    public let weight: Int
    public let petDescription: String
    public let petImage: String
    public let vaccines: [String]
    public let owner: Owner

    enum CodingKeys: String, CodingKey {
        case id, name, breed, sex, age, color, weight
        case petDescription = "description"
        case petImage = "pet_image"
        case vaccines, owner
    }
}

File: android/Pets/app/src/main/java/com/pets/models/Pet.kt
package com.pets.models

data class Pet(val id: Int,
               val name: String,
               val breed: String,
               val sex: String,
               val age: Int,
               val color: String,
               val weight: Int,
               val petDescription: String,
               val petImage: String,
               val vaccines: ArrayList<String>,
               val owner: Owner)